tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go

```diff
@@ -488,6 +493,46 @@ func (a *AwsProvider) createInstance(imageID, instanceType, sshKey string,
 	return runResult.Instances[0], nil
 }
 
+// createOpenShiftTagSpecification creates TagSpecification object to add OpenShift tag while
+// launching an instance.
+func (a *AwsProvider) createOpenShiftTagSpecification() ([]*ec2.TagSpecification, error) {
+	infraID, err := a.GetInfraID()
+	if err != nil {
+		return nil, fmt.Errorf("error getting infrastructure ID for the OpenShift cluster, %v", err)
+	}
+	tagslist := map[string]string{
+		// Add OpenShift tag, so that
+		// - The kubelet can communicate with cloud provider
+		// - TearDown & Reaper job in OpenShift CI can delete the virtual machine as part of cluster
+		infraIDTagKeyPrefix + infraID: infraIDTagValue}
+
+	awsTagslist, err := createTagList(tagslist)
+	if err != nil {
+		return nil, fmt.Errorf("error creating tags %v", err)
+	}
+
+	tagspec := &ec2.TagSpecification{
+		ResourceType: aws.String(ec2.ResourceTypeInstance),
+		Tags:         awsTagslist,
+	}
+
+	return []*ec2.TagSpecification{tagspec}, nil
+}
+
+// createTagList is a generic function to create ec2.Tag object list given
+// a map of required tags
+func createTagList(taglist map[string]string) ([]*ec2.Tag, error) {
```

Comment:
I believe you're having a map here for future expansion. As of now, to me it looks like we just pass a tuple(k, v) as we don't have too many tags being attached and this function is always called on a single tag.

Reply:
Yes. This function can take multiple tags. Even if we pass a tuple, we need to return list of form []*ec2.Tag. Hence, I kept it as a map.

Reply:
s/taglist/tags
s/tags/ec2Tags